provider/overlays.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provider
+
+import (
+	_ "embed" // Needed to support go:embed directive
+	"fmt"
+	"maps"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+const (
+	callbackFunctionTok = "aws:lambda/callbackFunction:CallbackFunction"
+	codePathOptionsTok  = "aws:lambda/CodePathOptions:CodePathOptions"
+)
+
+var arnToken = fmt.Sprintf("#/types/%s", awsTypeDefaultFile(awsMod, "ARN"))
+
+// resourceOverlays augment the resources defined by the upstream AWS provider
+var resourceOverlays = map[string]schema.ResourceSpec{}
+
+// typeOverlays augment the types defined by the upstream AWS provider
+var typeOverlays = map[string]schema.ComplexTypeSpec{}
+
+//go:embed pkg/overlays/examples/callbackFunction.md
```

Comment:
This is a moot point for just one file but if we're doing it a lot we can avoid shipping this into pulumi-resource-aws binary to avoid bulking it up. Trouble with resources.go code is that it's evaluated twice, once for tfgen and once for runtime startup in the provider, so it's a little tricky, but can be done with os.ReadFile that only runs for `make tfgen` case.

Reply:
IMO we should opt for the simpler option here to avoid bugs and trouble down the road. Even if we had multiple files it wouldn't be a lot of storage size we're adding. The markdown file for the callbackFunction has ~9KB, so we'd need a lot of them before this has meaningful impact on the overall size of the provider (~794MB right now).

Reply:
You are right, I concede the point.. The provider binary is too large but we need to look holistically as to what to do with that, this is not really the root problem here.